### PR TITLE
Credit card unit testcases

### PR DESCRIPTION
**Pull request by KajalMathur**
Kindly review my CR.

### CODE REVIEW

File: src/test/java/com/customer/customerService/CrediCardApplicationTest.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import org.junit.Before;
+import org.junit.Test;
+
+import com.customer.model.CreditCard;
+import com.customer.repository.CreditCardRepository;
+import com.customer.service.CreditCardServiceImpl;
+
+
+
+public class CrediCardApplicationTest {
```
Comment (AakashIsTheLimit) [inline src/test/java/com/customer/customerService/CrediCardApplicationTest.java]: soo many blank lines on top... limit it to 1

File: src/test/java/com/customer/customerService/CrediCardApplicationTest.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	
+	CreditCard creditCardData, creditCardDataWithNull, creditCardData1, creditCardData2, creditCardData3, creditCardData4;
+	
+	List<CreditCard> creditCardList;
+	
+	CreditCardRepository creditCardRepositoryMock;
+	
+	CreditCardServiceImpl creditCardServiceImpl;
+	
+	List<CreditCard> creditCardList1;
```
Comment (AakashIsTheLimit) [inline src/test/java/com/customer/customerService/CrediCardApplicationTest.java]: Try to use ImmutableList for tests... read more for the reason why

File: src/test/java/com/customer/customerService/CrediCardApplicationTest.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	 
+	 @Test(expected = IllegalArgumentException.class)
+	 public void verifyCreditCardPostwithNull() {
+	        when(creditCardRepositoryMock.save(null)).thenThrow(IllegalArgumentException.class);
+	        creditCardServiceImpl.createCreditCardInfo(null);
+	 }
+	
+	 @Test
+	 public void verifyCreditCardPostWithoneNullValue() {
+		 when(creditCardRepositoryMock.save(creditCardDataWithNull)).thenReturn(creditCardDataWithNull);
```
Comment (AakashIsTheLimit) [inline src/test/java/com/customer/customerService/CrediCardApplicationTest.java:116]: indentation

File: src/test/java/com/customer/customerService/CrediCardApplicationTest.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+		creditCardList1 = new ArrayList<CreditCard>();
+		creditCardList1.add(creditCardData);
+		
+		creditCardList2 = new ArrayList<CreditCard>();
+		creditCardList2.add(creditCardData);
+		creditCardList2.add(creditCardData1);
+		creditCardList2.add(creditCardData2);
+		creditCardList2.add(creditCardData3);
+		creditCardList2.add(creditCardData4);
+
```
Comment (AakashIsTheLimit) [inline src/test/java/com/customer/customerService/CrediCardApplicationTest.java]: why is there need of creditCardList2 .. seems same as creditCardList

File: src/test/java/com/customer/customerService/CrediCardApplicationTest.java (hunk 1 above)
Comment (KajalMathur) [reply src/test/java/com/customer/customerService/CrediCardApplicationTest.java]: Done.

File: src/test/java/com/customer/customerService/CrediCardApplicationTest.java (hunk 2 above)
Comment (KajalMathur) [reply src/test/java/com/customer/customerService/CrediCardApplicationTest.java]: Done.

File: src/test/java/com/customer/customerService/CrediCardApplicationTest.java (hunk 3 above)
Comment (KajalMathur) [reply src/test/java/com/customer/customerService/CrediCardApplicationTest.java:116]: Done.

File: src/test/java/com/customer/customerService/CrediCardApplicationTest.java (hunk 4 above)
Comment (KajalMathur) [reply src/test/java/com/customer/customerService/CrediCardApplicationTest.java]: Done.

File: src/test/java/com/customer/customerService/CrediCardApplicationTest.java [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+	 
+	 @Test(expected = IllegalArgumentException.class)
+	 public void verifyCreditCardPostwithNull() {
+	        when(creditCardRepositoryMock.save(null)).thenThrow(IllegalArgumentException.class);
+	        creditCardServiceImpl.createCreditCardInfo(null);
+	 }
+	
+	 @Test
+	 public void verifyCreditCardPostWithoneNullValue() {
+		 when(creditCardRepositoryMock.save(creditCardDataWithNull)).thenReturn(creditCardDataWithNull);
```
Comment (AakashIsTheLimit) [inline src/test/java/com/customer/customerService/CrediCardApplicationTest.java:116]: why is it still not correct indentation ?

**Review by AakashIsTheLimit — APPROVED** [review]

**Pull request merged**